### ISSUE TITLE
Two users with the same email

### ISSUE THREAD

**Issue by angaither** [DoSomething/northstar#80]
![](https://slack-files.com/files-pub/T024GV2BW-F04FS6MDJ-5559b52b65/screen_shot_2015-04-21_at_1.42.32_pm.png)

**Comment by angaither** [comment]
This seems to happen when the user is forwarded into Drupal.  Or when the user is attempted to be forwarded, since locally, not all devs have a local instance running.

@jonuy what are your thoughts on [commenting out this block of code](https://github.com/DoSomething/northstar/blob/dev/app/controllers/UserController.php#L67-L79) 

We can't seem to come up with a logic of when we need Drupal users, and I'm not sure we need it upon registration....

**Comment by jonuy** [comment]
@angaither Shouldn't this block still at least catch it and use the already existing user instead of creating a new one?

```
if (Input::has('email')) {
  $user = User::where('email', '=', $check['email'])->first();
} elseif (Input::has('mobile')) {
  $user = User::where('mobile', '=', $check['mobile'])->first();
}
```

And to the question on forwarding the registration - general rule I'm going by is that a Drupal user needs to get created when there's an assumption of cross platform support. So probably not on AGG/CGG, but mobile app - yea. 

I think it's ok for our separate apps to make that decision on a per-app basis. So that block of you code you mentioned, instead of checking `!$user->drupal_id`, what do you think about a flag specifying that the reg event should forwarded to Drupal? ie - `?fwd_to_drupal=true` or something.

The other way to approach this would be to make Drupal forward registration and login events to Northstar. This doesn't seem like a trivial thing though. Or is it?

**Comment by angaither** [comment]
@jonuy that block you mentioned is trying to find and use existing the user. 
right below that block is this
```
  if (!$user) {
      $user = new User;
``
otherwise it uses the person it found in the query to update the existing user

yes, I think we need a `fwd_to_drupal` rule. But maybe that lives with the `signup` & `reportback` logic
...since those are the cases where we need a drupal user...

Forwarded Drupal users -> northstar is already something on the list of things to build
https://github.com/DoSomething/dosomething/issues/4276

**Comment by jonuy** [comment]
Right, so I guess I'm not getting why a user wasn't found in the `User::where` queries.

DoSomething/dosomething#4276 sounds _beautiful_. Yea, if that's happening, then the signup and reportback are the only ones that actually need the drupal user.

I guess a heads up for #62 then is that we'll need to possibly create a drupal user in that signup or reportback event.

**Comment by angaither** [comment]
the duplicate return I think it what was causing this. I pushed the wrong branch, but this is now fixed with #87
